pkg/publish/cmd.go

```diff
@@ -84,6 +85,8 @@ func init() {
 		"The file containing a github token.")
 	publishCmd.PersistentFlags().StringVar(&flags.grafanatoken, "grafanatoken", flags.grafanatoken,
 		"The file containing a grafana.com API token.")
+	publishCmd.PersistentFlags().StringVar(&flags.cosignkey, "cosignkey", flags.cosignkey,
```

Comment:
for github token we use a file. Should we do similar here? or read from env?

Reply:
I guess if `cosign` takes it as a flag so can we

Reply:
+1 for using a file. Using env is dangerous as we can leak its content as it's passed as CLI arguments to commands

Reply:
Looking at the cosign command, `-key` is `path to the private key file, KMS URI or Kubernetes Secret` so not the actual key. So as discussed in the T&R WG call, the path will end up being output in the `util.VerboseCommand`, and not the actual key. I would like to have Path in the variable name so we understand that, but cosign uses the `-key` flag.

Reply:
Right, the expected value here (in the case of Prow) is a GCP KMS key of format `gcpkms://projects/host-project/locations/global/keyRings/keyring-name/cryptoKeys/key-name/versions/1`, as documented in https://github.com/sigstore/cosign/blob/main/KMS.md. That *path* is stored in the prow cluster in a secret, and is exposed as an environment variable via the kubernetes secrets api in istio/test-infra/pull/3573.

If you guys think it would be better to instead expose it as a file and then read it in the go code I can do that as well.